### ISSUE TITLE
Engine roll appears to be causing a failure in mac_android_android_semantics_integration_test

### ISSUE THREAD

**Issue by zanderso** [flutter/flutter#83751]
https://ci.chromium.org/ui/p/flutter/builders/prod/Mac_android%20android_semantics_integration_test/1367/overview

on the roll here

https://github.com/flutter/flutter/pull/83742

Maybe from? https://github.com/flutter/engine/pull/25373 @chunhtai

**Comment by chunhtai** [comment]
It is also possible https://github.com/flutter/engine/pull/26515/files cc @gaaclarke

**Comment by gaaclarke** [comment]
@chunhtai has a plausible explanation where the StandardMessageCodec handles byte[] type but not ByteBuffer type.  This interferes with how the StandardMethodCodec is using it.  I've never seen how the ByteBuffer can get passed directly to the StandardMessageCodec but it is a likely explanation.

**Comment by iskakaushik** [comment]
From @jason-simmons 

```
It looks like https://github.com/flutter/engine/commit/9502b6f0d292c8471e3ae2e7e220840e018e0b4e is returning a SpannableString that is accessed by the code in the test that responds to a platform message method call.

StandardMessageCodec does not have a case for SpannableString or CharSequence
```

**Comment by iskakaushik** [comment]
The revert of the a11y PR has landed. De-escalating it to P3. Assigning to @chunhtai to track re-landing his PR.
